Yoast/wordpress-seo
# Rename indexation routes and options

## Context


* **NB: If this PR is not merged before 15.3, some version numbers need to be changed in the code** (including in the upgrade routine).

## Summary


This PR can be summarized in the following changelog entry:

* Changes `indexation` to `indexing` in two options and all the endpoints.

## Relevant technical choices:

* I wrote upgrade routines for the `indexing_started` and `indexables_indexing_completed` options.

## Test instructions

This PR can be tested by following these steps:

The **CR** is easiest to do by commit.

**Test the option renames**
* Reset the indexables tables using the Yoast Test Helper.
* See that `wp_options` --> `wpseo` holds the options `indexation_started` and `indexables_indexation_completed`.
* Using the Test Helper, set the plugin version to 15.0.
* The upgrade routine is now executed.
* See that the option keys have been renamed to `indexing_started` and `indexables_indexing_completed`.
* Their values should still be the same.

**Test the route renames and the prepare deprecation**
* Reset the indexables.
* Run the indexing process on the Tools page with your network tab open.
* See that the `prepare` action is being executed. 
* When you hoover over the actions in the Network tab, see that the namespace contains `indexing` rather than `indexation`.


### Test instructions for QA when the code is in the RC



QA can test this PR by following these steps:

* When testing this PR, [please use version 1.9 or earlier of the Yoast Test Helper plugin](https://github.com/Yoast/yoast-test-helper/releases/tag/1.9.0).
  * Using the latest test helper to reset the indexables tables unfortunately interferes with testing the upgrade routine.
* Use the steps as described in the test instructions above.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* The complete indexing process: resetting, indexing, stopping halfway through, everything okay upon refresh after indexing, the notification.

## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/P2-411
